include/swift/Parse/Lexer.h

```diff
@@ -282,17 +280,16 @@ class Lexer {
   }
 
   State getStateForEndOfTokenLoc(SourceLoc Loc) const {
-    return State(getLocForEndOfToken(SourceMgr, Loc), LeadingTrivia,
-                 TrailingTrivia);
+    return State(getLocForEndOfToken(SourceMgr, Loc));
   }
 
   /// \brief Restore the lexer state to a given one, that can be located either
   /// before or after the current position.
   void restoreState(State S, bool enableDiagnostics = false) {
     assert(S.isValid());
     CurPtr = getBufferPtrForSourceLoc(S.Loc);
-    LeadingTrivia = S.LeadingTrivia;
-    TrailingTrivia = S.TrailingTrivia;
+    LeadingTrivia.clear();
+    TrailingTrivia.clear();
```

Comment:
Are these two lines necessary at all? `LexImpl` will set them, right?

Reply:
It's necessary, `lexImpl()` only *append* to them.
Probably, we should put this to the top of `lexImpl()` i.e:

```swift
void Lever::lexImpl() {
  LeadingTrivia.clear()
  TrailingTrivia.clear()
  ...
```
We could do that in another PR?

Reply:
We certainly could. Feel free to merge:)